x-pack/plugins/security_solution/scripts/extract_tactics_techniques_mitre.js

```diff
@@ -67,98 +71,124 @@ const getSubtechniquesOptions = (subtechniques) =>
 }`.replace(/(\r\n|\n|\r)/gm, ' ')
   );
 
-const getIdReference = (references) =>
-  references.reduce(
-    (obj, extRef) => {
-      if (extRef.source_name === 'mitre-attack') {
-        return {
-          id: extRef.external_id,
-          reference: extRef.url,
-        };
+const getIdReference = (references) => {
+  const ref = references.find((r) => r.source_name === 'mitre-attack');
+  if (ref != null) {
+    return {
+      id: ref.external_id,
+      reference: ref.url,
+    };
+  } else {
+    return { id: '', reference: '' };
+  }
+};
+
+const extractTacticsData = (mitreData) => {
+  const tactics = mitreData
+    .filter((obj) => obj.type === 'x-mitre-tactic')
+    .reduce((acc, item) => {
```

Comment:
nit: by moving filter condition inside `reduce`, could save one traverse through tactics

Reply:
Let's address it as part of a follow-up ticket.